src/seb/registered_models/llm2vec_models.py

```diff
@@ -0,0 +1,205 @@
+        return task_name_to_instruct.get(task.name, default_instruction)
+    if task.task_type in ["Reranking"]:
+        return "Retrieve semantically similar passages."
+    if task.task_type in ["Retrieval"]:
+        task_name_to_instruct: dict[str, str] = {
+            "Twitterhjerne": "Retrieve answers to questions asked in Danish tweets",
+            "SwednRetrieval": "Given a Swedish news headline retrieve summaries or news articles",
+            "TV2Nord Retrieval": "Given a summary of a Danish news article retrieve the corresponding news article",
+            "DanFEVER": "Given a claim in Danish, retrieve documents that support the claim",
+            "SNL Retrieval": "Given a lexicon headline in Norwegian, retrieve its article",
+            "NorQuad": "Given a question in Norwegian, retrieve the answer from Wikipedia articles",
+            "SweFAQ": "Retrieve answers given questions in Swedish",
+            "ArguAna": "Given a claim, find documents that refute the claim",
+            "ClimateFEVER": "Given a claim about climate change, retrieve documents that support or refute the claim",
+        }
+        default_instruction = "Retrieve text based on user query."
+        return task_name_to_instruct.get(task.name, default_instruction)
+    return ""
+
+
+def batched(iterable: Iterable[T], n: int) -> Iterable[tuple[T, ...]]:
+    if n < 1:
+        raise ValueError("n must be at least one")
+    it = iter(iterable)
+    while batch := tuple(islice(it, n)):
+        yield batch
+
+
+# Custom SEB Encoder Model
+class LLM2VecModel(Encoder):
+    def __init__(
+        self,
+        base_model_name_or_path: str,
+        peft_model_name_or_path: str,
+        max_length: int,
+        max_batch_size: Optional[int] = None,
+        **kwargs: Any,
+    ):
+        logger.info("Started loading LLM2Vec model")
+        try:
+            from llm2vec import LLM2Vec
+        except ImportError:
+            raise ImportError(
+                "To use the LLM2Vec models `llm2vec` is required. Please install it with `pip seb[llm2vec]."
+            )
+        extra_kwargs = {}
+        try:
+            import flash_attn  # noqa
+
+            extra_kwargs["attn_implementation"] = "flash_attention_2"
+        except ImportError:
+            logger.warning(
+                "LLM2Vec models were trained with flash attention enabled. For optimal performance, please install the `flash_attn` package"
+            )
+        self.model = LLM2Vec.from_pretrained(
+            base_model_name_or_path=base_model_name_or_path,
+            peft_model_name_or_path=peft_model_name_or_path,
+            **extra_kwargs,
+            **kwargs,
+            device_map="auto",
+            torch_dtype=torch.bfloat16,
+        )
+        self.max_length = max_length
+        self.max_batch_size = max_batch_size
+
+    def preprocess(self, sentences, instruction) -> Union[List[str], List[List[str]]]:
```

Comment:
type hint me
```suggestion
    def preprocess(self, sentences, instruction) -> list[str] | list[list[str]]:
```
You can add 
```from __future__ import annotations```